tests/program/test_extract.py

```diff
@@ -293,3 +293,48 @@ def test_program_extract_single_resource_csv():
         json.dumps(actual.stdout, ensure_ascii=False)
         == '"id,name\\n1,中国人\\n2,english\\n"'
     )
+
+
+# Bugs
+
+
+def test_extract_resource_from_csv_semicolon_delimiter_issue_1009():
+    actual = runner.invoke(program, "extract data/issue-1009-semicolon.csv --csv")
+    assert actual.exit_code == 0
+    assert (
+        json.dumps(actual.stdout, ensure_ascii=False)
+        == '"fieldNameA;fieldNameB\\n0123;c\\n"'
+    )
+
+
+def test_extract_resource_from_csv_comma_delimiter_issue_1009():
+    actual = runner.invoke(program, "extract data/issue-1009-comma.csv --csv")
+    assert actual.exit_code == 0
+    assert (
+        json.dumps(actual.stdout, ensure_ascii=False)
+        == '"fieldNameA,fieldNameB\\n123,c\\n"'
+    )
+
+
+def test_extract_resource_from_csv_semicolon_delimiter_param_issue_1009():
+    actual = runner.invoke(
+        program,
+        'extract data/issue-1009-semicolon.csv --dialect \'{"delimiter": ";"}\' --csv',
+    )
+    assert actual.exit_code == 0
+    assert (
+        json.dumps(actual.stdout, ensure_ascii=False)
+        == '"fieldNameA;fieldNameB\\n123;c\\n"'
+    )
+
+
+def test_extract_resource_from_csv_comma_delimiter_param_issue_1009():
+    actual = runner.invoke(
+        program,
+        'extract data/issue-1009-comma.csv --dialect \'{"delimiter": ","}\' --csv',
```

Comment:
Unfortunately, we can't do something like this. `extract --dialect` means "specify a dialect of the input file" but not for the output one.